### PR TITLE
Add progress dashboard; fixes #34

### LINKED ISSUE #34: [Model Dashboard] As a model administrator, I want to know the progress of the project
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
As a model administrator, I want to know the progress of the project with all values published / not
published + how many are still empty
Need to try with different solutions to find the best :
- Pie chart 
- bar chart

- Percentage 
- numbers

### Steps to Reproduce
NA
### Logs
NA

### System Configuration
NA

### PR DESCRIPTION
**Pull request by justineatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Add 4 versions of dashboard for 3 types of values : 
- published/publishable values 
- default / other values for actual values
- default / other values for published values
Each graph has different features.
I tried to arrange pages to propose ESA and let them choose what they want.

### CODE REVIEW

File: COMETwebapp/Components/CloseModelButton.razor [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+    Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see http://www.gnu.org/licenses/.
+------------------------------------------------------------------------------->
+
+@inject ISessionAnchor SessionAnchor
+@inject NavigationManager NavigationManager
```
Comment (lxatrhea) [inline COMETwebapp/Components/CloseModelButton.razor:24]: Just out of curiosity: From an architectural point of view, why do you use an interface for SessionAnchor, but not for the NavigationManager?

**Review by lxatrhea — APPROVED** [review]

File: COMETwebapp/Components/DefaultValues.razor [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    /// </summary>
+    List<DataChart> Data = new List<DataChart>();
+
+    /// <summary>
+    /// Get all parameter values and build data for the graphs
+    /// </summary>
+    protected override void OnInitialized()
+    {
+        if(ISessionAnchor.OpenIteration != null){
+            var listDomains = ISessionAnchor.Session.QueryDomainOfExpertise(ISessionAnchor.OpenIteration).ToList();
```
Comment (samatrhea) [inline COMETwebapp/Components/DefaultValues.razor:56]: no need to call this "listDomains", just call it "domains"

File: COMETwebapp/Components/DefaultValues.razor [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            List<ParameterValueSet> listModelValueSet = ISessionAnchor.GetParameterValueSets();
```
Comment (samatrhea) [inline COMETwebapp/Components/DefaultValues.razor:57]: please rename this to `parameterValueSets`

File: COMETwebapp/Components/DxChart/ChartBarSeries.razor [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                      AggregationMethod="Enumerable.Sum" />
+    <DxChartBarSeries Name="@NameSerie2" Filter="@((DataChart s) => s.Argument.Equals(NameSerie2))"
+                      ArgumentField="@(s => s.Domain)" ValueField="@(s => s.Value)"
+                      AggregationMethod="Enumerable.Sum" />
+    <DxChartLegend Position="RelativePosition.Outside" HorizontalAlignment="HorizontalAlignment.Right" />
+</DxChart>
+
+@code {
+    /// <summary>
+    /// Data represented in the chart bar
```
Comment (samatrhea) [inline COMETwebapp/Components/DxChart/ChartBarSeries.razor:42]: this is typically called a "bar chart" and not a "chart bar"

File: COMETwebapp/Components/DxChart/ChartBarSeries.razor [hunk 5]
```diff
@@ -0,0 +1,64 @@
+﻿<!------------------------------------------------------------------------------
```
Comment (samatrhea) [inline COMETwebapp/Components/DxChart/ChartBarSeries.razor:1]: i would not make a folder or namespace called DxCharts. The fact that we use DevExpress is not relevant and i don;t think it should be part of the name. Either invent a new name that says something about why and how the controls or used, or delete the DxChart folder and move all controls to the Components folder.

File: COMETwebapp/Components/ParameterTable.razor [hunk 6]
```diff
@@ -52,25 +52,33 @@ Copyright (c) 2022 RHEA System S.A.
...
-        foreach(var element in ISessionAnchor.OpenIteration.Element)
+        var ListElements = ISessionAnchor.OpenIteration.Element;
+        if (ListElements.Count != 0)
         {
-            foreach(var parameter in element.Parameter)
+            foreach (var element in ListElements)
             {
-                foreach (var valueset in parameter.ValueSet) {
-                    Parameters.Add(new BuiltParameter()
+                var ListParameters = element.Parameter;
```
Comment (samatrhea) [inline COMETwebapp/Components/ParameterTable.razor:60]: please do not use the word `list` in variable names, `parameters` is good enough

File: COMETwebapp/Components/PublishedValues.razor [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    /// </summary>
+    List<DataChart> Data = new List<DataChart>();
+
+    /// <summary>
+    /// Get all parameter values and build the list of data 
+    /// </summary>
+    protected override void OnInitialized()
+    {
+        if(ISessionAnchor.OpenIteration != null){
+            var listDomains = ISessionAnchor.Session.QueryDomainOfExpertise(ISessionAnchor.OpenIteration).ToList();
```
Comment (samatrhea) [inline COMETwebapp/Components/PublishedValues.razor:50]: please rename to `domains`

File: COMETwebapp/Components/PublishedValues.razor [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            List<ParameterValueSet> listModelValueSet = ISessionAnchor.GetParameterValueSets();
```
Comment (samatrhea) [inline COMETwebapp/Components/PublishedValues.razor:51]: please rename to `parameterValueSets`

File: COMETwebapp/Pages/ModelDashboard.razor [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    /// Update the component depending on the selected <see cref="DataChoice"/>
+    /// </summary>
+    /// <param name="e">Event raised when any change appears in the select component</param>
+    public void UpdateChoice(ChangeEventArgs e)
+    {
+        Console.WriteLine(e.Value.ToString());
+        if (e.Value.ToString().Equals("defaultActualValues"))
+        {
+            Choice = DataChoice.DefaultActualValues;
+            base.StateHasChanged();
```
Comment (samatrhea) [inline COMETwebapp/Pages/ModelDashboard.razor:75]: i think ``base` can be renamed to `this`

File: COMETwebapp/SessionManagement/ISessionAnchor.cs [hunk 10]
```diff
@@ -98,6 +98,8 @@ public interface ISessionAnchor
         /// <returns>
         /// A container of <see cref="DomainOfExpertise"/>
         /// </returns>
-        IEnumerable<DomainOfExpertise> GetModelDomains(EngineeringModelSetup modelSetup);
+        IEnumerable<DomainOfExpertise> GetModelDomains(EngineeringModelSetup? modelSetup);
+
+        List<ParameterValueSet> GetParameterValueSets();
```
Comment (samatrhea) [inline COMETwebapp/SessionManagement/ISessionAnchor.cs:103]: missing docs

File: COMETwebapp/SessionManagement/SessionAnchor.cs [hunk 11]
```diff
@@ -147,18 +145,18 @@ public IEnumerable<EngineeringModelSetup> GetParticipantModels()
...
-                       yield return domain;
-                    }
-                }
-            }
+            var domains = new List<DomainOfExpertise>();
+            modelSetup?.Participant.FindAll(p => p.Person.Name.Equals(this.Session.ActivePerson.Name)).ForEach(p => p.Domain.ForEach(d => domains.Add(d)));
+            return domains.DistinctBy(d => d.Name).OrderBy(d => d.Name);
+        }
+
+        public List<ParameterValueSet> GetParameterValueSets()
```
Comment (samatrhea) [inline COMETwebapp/SessionManagement/SessionAnchor.cs:155]: missing docs

File: COMETwebapp/SessionManagement/SessionAnchor.cs [hunk 12]
```diff
@@ -147,18 +145,18 @@ public IEnumerable<EngineeringModelSetup> GetParticipantModels()
         /// <returns>
         /// A container of <see cref="DomainOfExpertise"/>
```
Comment (samatrhea) [inline COMETwebapp/SessionManagement/SessionAnchor.cs:146]: can you explain in words in the docs what Domains of Expertise are returned?

**Review by samatrhea — CHANGES_REQUESTED** [review]

**Pull request closed without merging**